reana-ui/src/pages/workflowDetails/components/filePreview/FilePreview.js

```diff
@@ -0,0 +1,213 @@
+  return (
+    <Modal.Content className={styles["fill-modal"]}>
+      <object data={url}>
+        <Message icon info style={{ margin: "1.5rem", width: "auto" }}>
+          <Icon name="info circle" />
+          <Message.Content>
+            Click{" "}
+            <a href={url} target="_blank" rel="noopener noreferrer">
+              {fileName}
+            </a>{" "}
+            to open the PDF file, or use the download button.
+          </Message.Content>
+        </Message>
+      </object>
+    </Modal.Content>
+  );
+}
+
+const PREVIEW_MIME_PREFIX_WHITELIST = {
+  "image/": ImagePreview,
+  "text/html": HTMLPreview,
+  "text/": TextPreview,
+  "application/json": TextPreview,
+  "application/pdf": PDFPreview,
+  "application/x-root": ROOTPreview,
+};
+
+const FILE_SIZE_LIMIT = 5 * 1024 ** 2; // 5MB
+
+/**
+ * Check if the given file name matches any given mime-type
+ * @param {Array} list Array of mime-types to check against
+ * @param {String} fileName File name to check
+ * @return {Boolean} Extension that matches the file name
+ */
+function matchesMimeType(list, fileName) {
+  const mimeType = getMimeType(fileName);
+  return mimeType && list.find((ext) => mimeType.startsWith(ext));
+}
+
+/**
+ * Verify if file overpasses size limit or has a blacklisted mime-type.
+ * @param {String} fileName File name
+ * @param {Integer} size File size
+ * @return {component.Message|null} Component displaying reason or null
+ */
+function checkConstraints(fileName, size) {
+  let content;
+  const match = matchesMimeType(
+    Object.keys(PREVIEW_MIME_PREFIX_WHITELIST),
+    fileName
+  );
+  if (!match) {
+    const fileExt = fileName.split(".").pop();
+    content = `${fileExt} files cannot be previewed. Please use download.`;
+  } else if (size > FILE_SIZE_LIMIT) {
+    content = `File size is too big to be previewed (limit ${
+      FILE_SIZE_LIMIT / 1024 ** 2
+    }MB). Please use download.`;
+  }
+  return content ? <Message icon="info circle" content={content} info /> : null;
+}
+
+/**
+ * Build the URL of a file.
+ */
+function getFileURL(workflow, fileName, preview = true) {
+  return WORKFLOW_FILE_URL(workflow, fileName, { preview });
+}
+
+export default function FilePreview({ workflow, fileName, size, onClose }) {
+  let preview = null;
+  // Check whether the file can be previewed
+  const message = checkConstraints(fileName, size);
+  if (message) {
+    preview = <Modal.Content scrolling>{message}</Modal.Content>;
+  } else {
+    const mimeType = matchesMimeType(
+      Object.keys(PREVIEW_MIME_PREFIX_WHITELIST),
+      fileName
+    );
+    const Preview = PREVIEW_MIME_PREFIX_WHITELIST[mimeType];
+    preview = (
+      <Preview
+        workflow={workflow}
+        fileName={fileName}
+        size={size}
+        url={getFileURL(workflow, fileName)}
+      />
+    );
+  }
+
+  return (
+    <Modal open closeIcon onClose={onClose ? onClose : () => {}}>
+      <Modal.Header>{fileName}</Modal.Header>
+      <Suspense
```

Comment:
Since, according to the docs, "Only Suspense-enabled data sources will activate the Suspense component", such as the components loaded with `lazy` (see [the note in this page](https://react.dev/reference/react/Suspense#:~:text=Show%20less-,Note,-Only%20Suspense%2Denabled)), maybe it makes more sense to move this inside the `ROOTPreview` component. Another option would be to create some sort of wrapper around the `_request` method to make it communicate with `Suspense`, but that's something we can maybe address in another issue. What do you think?

Reply:
It's true that only `ROOTPreview` uses it for now, but future previewer might be lazy loaded for the same reason as jsroot (i.e. big dependency). For this reason, I prefer having `Suspense` outside of `ROOTPreview`, so that the fallback `Loader` is defined only once (and not into every lazy loaded component).

Regarding `_request` being "Suspense-enabled", I think that might be a cool feature, but that needs quite a lot of work so it's probably better to investigate and implement it in another issue/PR.